### PR TITLE
[CI] Restore the exported documentation linter

### PR DESCRIPTION
**Pull request by amenasria**
### What does this PR do?

This PR fixes golangci linter errors by excluding the remaining errors.

### Motivation

We used to enforce that exported functions and types are documented. This was done through `revive` but the check was disabled by default when we switched to `golangcli-lint`. Now that we tried to re-enable it we got a lot of errors. This PR addresses these errors by excluding them in the `.golangci.yml` file.

Some `revive` linter errors were already fixed in these PRs for some teams:
- Serverless: https://github.com/DataDog/datadog-agent/pull/12158 
- Agent Core: https://github.com/DataDog/datadog-agent/pull/12134 
- Container Integrations: https://github.com/DataDog/datadog-agent/pull/12135 
- Remote Config: https://github.com/DataDog/datadog-agent/pull/12157 
- Agent Security: https://github.com/DataDog/datadog-agent/pull/12142

But they're still some errors and it would need a lot of time to fix them. The goal is to re-enable the linter as soon as possible to keep people from adding new linter errors in the future. Meanwhile, the teams will be able to remove the error exclusion (in the `.golangci.yml` file) after fixing their code.
 
This is only one step before a bigger fix.

### Describe how to test/QA your changes

You can run `golangci` linter in the repo:

```
golangci-lint run ./...
```

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by albertvaka — APPROVED** [review]

File: .golangci.yml [hunk 1]
```diff
@@ -5,13 +5,24 @@ run:
...
   exclude:
+    - "`RandomString|TimeNowNano|InitLogging|SetHostname` is unused" # 'deadcode' errors in test/benchmarks/util.go
+    - "Error return value of `io.WriteString` is not checked" # 'errcheck' errors in tools/dep_tree_resolver/go_deps.go
+    - "Error return value of `pem.Encode` is not checked" # 'errcheck' errors in test/integration/utils/certificates.go
+    - "exported: exported type EasyjsonTime|NewEasyjsonTime|EasyjsonTime.MarshalEasyJSON|EasyjsonTime.UnmarshalJSON should have comment or be unexported" # 'revive' errors in pkg/security/utils/json.go
+    - "exported: exported const Exec should have comment \\(or a comment on this block\\) or be unexported" # 'revive' errors in pkg/process/events/model/model_common.go
+    - "exported: exported const APIName should have comment \\(or a comment on this block\\) or be unexported" # 'revive' errors in pkg/serverless/trace/inferredspan/constants.go
+    - "unnecessary conversion" # 'unconvert' errors in test/integration/utils/certificates_test.go
+    - "SA1019*" # 'staticcheck' errors about SA1019
+    - "SA4011*" # 'staticcheck' errors about SA4011
```
Comment (mx-psi) [inline .golangci.yml:24]: I think we should add this to the list for staticcheck checks at the very end of the file instead of here

Comment (mx-psi) [inline .golangci.yml:24]: I think we should add this to the list for staticcheck checks at the very end of the file instead of here

**Comment by amenasria** [comment]
A lot of linters errors are still not covered by the exclusions I added. Currently fixing that 🔧

**Comment by amenasria** [comment]
Kept fixing linters errors:

- #12352 
- #12353 
- #12354 
- #12356 
- #12357 

Waiting for these PRs to be merged to merge main into this branch ! 🦦

**Pull request merged**